src/nft.sol

```diff
@@ -13,185 +13,156 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-pragma solidity >=0.4.24;
+pragma solidity >=0.5.15 <0.6.0;
+pragma experimental ABIEncoderV2;
 
 import { ERC721Metadata } from "./openzeppelin-solidity/token/ERC721/ERC721Metadata.sol";
 import "./openzeppelin-solidity/cryptography/ECDSA.sol";
-import "./merkle.sol";
 
-contract AnchorLike {
-    function getAnchorById(uint) public view returns (uint, bytes32, uint32);
+contract AssetManagerLike {
+    function isAssetValid(bytes32 asset) external view returns (bool);
```

Comment:
Why not have a method `exists()` that reverts if it does not as opposed to having to check for a return value?

Reply:
I think the basic AssetRegistry implementation should look like this:
```
contract AssetRegistry {
    mapping (bytes32 => bool) public assets;
    function exists(bytes32 asset) public view {
        require(assets[asset], "asset-registry/not-found");
    }
}
```
Therefore the Abstract can look like this:

```
contract AbstractRegistryLike {
    function assets(bytes32) public view returns (bool);
    function exists(bytes32) public view;
}
```
This makes a check very simple:
```
registry.exists(asset);
```

Reply:
I think **read** operations should return a value or not. I believe the responsibility to react with an exception should be done outside of the read function itself.

Reply:
@lucasvo wdyt ?

Reply:
No, this is a commonly used pattern and leads to less code. We use this pattern in many places in our tinlake contracts. I've renamed the method to `verify` make it clear it is not used to read a value but simply verifies the asset.

```
contract AssetRegistry {
    mapping (bytes32 => bool) public assets;
    function verify(bytes32 asset) public view {
        require(assets[asset], "asset-registry/not-found");
    }
}

contract AbstractRegistryLike {
    function assets(bytes32) public view returns (bool);
    function verify(bytes32) public view;
}
```

Reply:
Solidity auto generates a read function already by having a `public` mapping:
```
    mapping (bytes32 => bool) public assets;
```

The solidity compiles this to the same bytecode as:
```
contract AssetRegistry {
    mapping (bytes32 => bool) assets;
    function assets(bytes32 item) public view returns (bool) {
        return assets[item];
    }
}
```
Writing a custom getter is redundant.